src/Entities/DreamStuff/DreamJellyfish.cs

```diff
@@ -0,0 +1,268 @@
+        public static readonly ParticleType[] P_DreamGlide = new ParticleType[CustomDreamBlock.DreamColors.Length];
+
+        // Could maybe use CustomDreamBlock.DreamParticle.
+        public struct DreamParticle {
+            public Vector2 Position;
+            public int Layer;
+            public Color EnabledColor, DisabledColor;
+            public float TimeOffset;
+        }
+        public DreamParticle[] Particles;
+        public static MTexture[] ParticleTextures;
+        public float Flash;
+
+        public Rectangle ParticleBounds = new Rectangle(-23, -35, 48, 42);
+
+        private readonly DreamDashCollider dreamDashCollider;
+        public bool AllowDreamDash {
+            get {
+                return dreamDashCollider.Active;
+            }
+            set {
+                dreamDashCollider.Active = value;
+            }
+        }
+
+        private DynData<Glider> gliderData;
+
+        public Sprite Sprite;
+        public MTexture CurrentFrame => Sprite.GetFrame(Sprite.CurrentAnimationID, Sprite.CurrentAnimationFrame);
+
+        public DreamJellyfish(EntityData data, Vector2 offset)
+            : this(data.Position + offset, data.Bool("bubble"), data.Bool("tutorial")) { }
+
+        public DreamJellyfish(Vector2 position, bool bubble, bool tutorial)
+            : base(position, bubble, tutorial) {
+            gliderData = new DynData<Glider>(this);
+
+            Sprite oldSprite = gliderData.Get<Sprite>("sprite");
+            Remove(oldSprite);
+            gliderData["sprite"] = Sprite = CommunalHelperModule.SpriteBank.Create("dreamJellyfish");
+            Add(Sprite);
+
+            Visible = false;
+
+            Add(dreamDashCollider = new DreamDashCollider(new Hitbox(28, 16, -13, -18), OnDreamDashExit));
+        }
+
+        public override void Awake(Scene scene) {
+            base.Awake(scene);
+
+            int w = ParticleBounds.Width;
+            int h = ParticleBounds.Height;
+            Particles = new DreamParticle[(int) (w / 8f * (h / 8f) * 1.5f)];
+            for (int i = 0; i < Particles.Length; i++) {
+                Particles[i].Position = new Vector2(Calc.Random.NextFloat(w), Calc.Random.NextFloat(h));
+                Particles[i].Layer = Calc.Random.Choose(0, 1, 1, 2, 2, 2);
+                Particles[i].TimeOffset = Calc.Random.NextFloat();
+
+                Particles[i].DisabledColor = Color.LightGray * (0.5f + Particles[i].Layer / 2f * 0.5f);
+                Particles[i].DisabledColor.A = 255;
+
+                Particles[i].EnabledColor = Particles[i].Layer switch {
+                    2 => Calc.Random.Choose(Calc.HexToColor("5b6ee1"), Calc.HexToColor("CC3B3B"), Calc.HexToColor("7daa64")),
+                    1 => Calc.Random.Choose(Calc.HexToColor("5fcde4"), Calc.HexToColor("7fb25e"), Calc.HexToColor("E0564C")),
+                    _ => Calc.Random.Choose(Calc.HexToColor("FFEF11"), Calc.HexToColor("FF00D0"), Calc.HexToColor("08a310")),
+                };
```

Comment:
you separated out the `CustomDreamBlock.DreamColors` but then didn't make use of them here